src/server/handlers/firestore/consents.validator.ts

```diff
@@ -0,0 +1,137 @@
+ Contributors
+ --------------
+ This is the official list of the Mojaloop project contributors for this file.
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Mojaloop Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+ * Mojaloop Foundation
+ - Name Surname <name.surname@mojaloop.io>
+
+ * Google
+ - Abhimanyu Kapur <abhi.kapur09@gmail.com>
+ --------------
+ ******/
+
+import { Consent } from '~/models/consent'
+
+/**
+ * Checks whether a consent document has all the necessary fields to perform
+ * a party lookup.
+ *
+ * @param consent the object representation of a consent that is stored
+ *                    on Firebase.
+ */
+export const isValidPartyLookup = (consent: Consent): boolean => {
+  if (
+    consent?.party?.partyIdInfo &&
+    consent.party.partyIdInfo.partyIdType &&
+    consent.party.partyIdInfo.partyIdentifier
+  ) {
+    return true
+  }
+  return false
+}
+
+/**
+ * Checks whether a consent document has all the necessary fields to be
+ * processed as an authenticated consent request.
+ *
+ * @param consent the object representation of a consent that is stored
+ *                    on Firebase.
+ */
+export const isValidAuthentication = (consent: Consent): boolean => {
+  if (
+    consent.consentRequestId &&
+    consent.consentId &&
+    consent.party &&
+    consent.party.partyIdInfo &&
+    consent.party.partyIdInfo.fspId &&
+    consent.initiatorId &&
+    consent.authChannels &&
+    consent.authToken
+  ) {
+    return true
+  }
+  return false
+}
+
+/**
+ * Checks whether a consent document has all the necessary fields to be
+ * processed as a consent request.
+ *
+ * @param consent the object representation of a consent that is stored
+ *                    on Firebase.
+ */
+export const isValidConsentRequest = (consent: Consent): boolean => {
+  if (
+    consent.authChannels &&
+    consent.scopes &&
+    consent.initiatorId &&
+    consent.party &&
+    consent.party.partyIdInfo?.fspId &&
+    consent.authUri
+  ) {
+    return true
+  }
+  return false
+}
+
+/**
+ * Checks whether a consent document has all the necessary fields to be
+ * processed as a request to generate challenge for a consent.
+ *
+ * @param consent the object representation of a consent that is stored
+ *                    on Firebase.
+ */
+export const isValidChallengeGeneration = (consent: Consent): boolean => {
+  if (consent.consentId && consent.party && consent.party.partyIdInfo?.fspId) {
```

Comment:
```suggestion
  return !!(consent?.consentId && consent.party?.partyIdInfo?.fspId)
```

Reply:
maybe returning expression will fit better in place of if/else -> true/false mapping?

Reply:
Oh right I think this is a better way to do it rather than using if/else or `!= null` :o